fw/t/unit/test_http2_parser_hpack.c

```diff
@@ -845,6 +840,37 @@ TEST(http2_parser_hpack, increment_all_static_indexes_for_request)
 		EXPECT_TFWSTR_EQ(&req->h_tbl->tbl[TFW_HTTP_HDR_RAW + 16],
 				 "via" "1.0 fred, 1.1 p.example.net");
 	}
+
+	FOR_REQ_H2_HPACK(
+	    HEADERS_FRAME_BEGIN();
+		// :method = "GET"
+		HEADER(INC_IND(INDEX(2), VALUE("GET")));
+		// :scheme = "https"
+		HEADER(INC_IND(INDEX(7), VALUE("https")));
+		// :path = "/"
+		HEADER(INC_IND(INDEX(4), VALUE("/")));
+	    HEADERS_FRAME_END();
+	)
+	{
+		EXPECT_EQ(req->method, TFW_HTTP_METH_GET);
+		EXPECT_TFWSTR_EQ(&req->h_tbl->tbl[TFW_HTTP_HDR_H2_SCHEME],
+				 ":scheme" "https");
+		EXPECT_TFWSTR_EQ(&req->uri_path, "/");
+	}
+
+	FOR_REQ_H2_HPACK(
+	    HEADERS_FRAME_BEGIN();
+		// :method = "GET"
+		HEADER(DYN_TBL_INDEX(1));
+		// :scheme = "https"
+		HEADER(DYN_TBL_INDEX(2));
+		// :path = "/"
+		HEADER(DYN_TBL_INDEX(3));
+	    HEADERS_FRAME_END();
+	)
+	{
+		EXPECT_EQ(req->method, TFW_HTTP_METH_GET);
+	}
```

Comment:
The test case is artificial. Index 2 in static table means `:method GET`, so while I didn't find explicit prohibition of sending `GET` value for index 2 in the RFC, I still don't think that we need to make our code to accept this kind of requests. I believe we must block such requests.

However, `$ ./h2spec http2/8.1.2.3 -t -h ubuntu -p 443 -k` (`ubuntu` is the name of my VM) raises the same warning. It seems this test https://github.com/summerwind/h2spec/blob/master/http2/8_1_2_1_pseudo_header_fields.go#L54 leads to the warning. Maybe because we have `TFW_TAG_HDR_H2_METHOD = TFW_TAG_HDR_H2_STATUS`. Need to investigate the problem more.

In general, we should keep our parser as strict as possible and allow only direct RFC requirements. I don't see in
>  If the header field name matches the header field name of an entry stored in the static table or the dynamic table, the header field name can be represented using the index of that entry.  In this case, the index of the entry is represented as an integer with a 6-bit prefix (see Section 5.1).  This value is always non-zero.

requirements to process the requests as in the test above. In my understanding this RFC citation is about name only indexes, e.g. 1.

We must satisfy common tools like `curl`, browsers, `h2spec` and so on. We should not process all imaginary requests, which aren't directly contradict RFCs. In this sense I believe this part of #1617 is incorrect.

Reply:
I agree. I've checked sources of nghttp2 and Golang std lib related to http2 and this test case is wrong.
Furthermore, I've found the reason of that warning. It has nothing to with method. As you've suggested, there was an issue with the HPACK hdr tags enum.